### PR TITLE
Cleanup vsphere_guest module

### LINKED ISSUE #24672: Connecting to v-center returns error that's not clear
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

vsphere_guest

##### ANSIBLE VERSION

```
AXSUNDHA-M-R10R:FreeBSD axsundha$ ansible --version
 [WARNING]: Optional dependency 'cryptography' raised an exception, falling back
to 'Crypto'

ansible 2.1.1.0
  config file = /Users/axsundha/ansible/Content_Security/FreeBSD/ansible.cfg
  configured module search path = Default w/o overrides

```

##### CONFIGURATION

inventory = /Users/axsundha/ansible/Content_Security/FreeBSD/hosts

##### OS / ENVIRONMENT
OS X EI Capiton

##### SUMMARY

When you run the playbook with vsphere_guest module, if the vmware URL is not complete, it returns mis-leading error

##### STEPS TO REPRODUCE

Run the playbook with the v-center and proper credentials and hostname

```
---
- name: Playbook for Content Security servers
  hosts: FreeBSD_VM

  roles:
    # Shutdown_Server
    # Shutdown_VM
    - Start_VM

---

- vsphere_guest:
    vcenter_hostname: https://<V-center hostname>
    username: <Username>
    password: <Password>
    guest: <Guest hostname>
    state: powered_on
    validate_certs: no
    force: yes

```

##### EXPECTED RESULTS

Turn on the VM

##### ACTUAL RESULTS

```
AXSUNDHA-M-R10R:FreeBSD axsundha$ ansible-playbook -vvv CS_FreeBSD.yaml
 [WARNING]: Optional dependency 'cryptography' raised an exception, falling back
to 'Crypto'

Using /Users/axsundha/ansible/Content_Security/FreeBSD/ansible.cfg as config file

PLAYBOOK: CS_FreeBSD.yaml ******************************************************
1 plays in CS_FreeBSD.yaml

PLAY [Playbook for Content Security servers] ***********************************

TASK [Start_VM : vsphere_guest] ************************************************
task path: /Users/axsundha/ansible/Content_Security/FreeBSD/roles/Start_VM/tasks/main.yml:3
<Guest hostname> ESTABLISH LOCAL CONNECTION FOR USER: axsundha
<Guest hostname> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1494941921.45-189921583388499 `" && echo ansible-tmp-1494941921.45-189921583388499="` echo $HOME/.ansible/tmp/ansible-tmp-1494941921.45-189921583388499 `" ) && sleep 0'
<Guest hostname> PUT /var/folders/br/2l8z6cfs03sfwszn020z2t5c0000gn/T/tmpRzc1us TO /Users/axsundha/.ansible/tmp/ansible-tmp-1494941921.45-189921583388499/vsphere_guest
<Guest hostname> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/local/bin/python2.7 /Users/axsundha/.ansible/tmp/ansible-tmp-1494941921.45-189921583388499/vsphere_guest; rm -rf "/Users/axsundha/.ansible/tmp/ansible-tmp-1494941921.45-189921583388499/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/var/folders/br/2l8z6cfs03sfwszn020z2t5c0000gn/T/ansible_2jRR_o/ansible_module_vsphere_guest.py", line 1879, in <module>
    main()
  File "/var/folders/br/2l8z6cfs03sfwszn020z2t5c0000gn/T/ansible_2jRR_o/ansible_module_vsphere_guest.py", line 1754, in main
    viserver.connect(vcenter_hostname, username, password)
  File "/Library/Python/2.7/site-packages/pysphere/vi_server.py", line 101, in connect
    request)._returnval
  File "/Library/Python/2.7/site-packages/pysphere/resources/VimService_services.py", line 2172, in RetrieveServiceContent
    response = self.binding.Receive(RetrieveServiceContentResponseMsg.typecode)
  File "/Library/Python/2.7/site-packages/pysphere/ZSI/client.py", line 540, in Receive
    self.ReceiveSOAP(**kw)
  File "/Library/Python/2.7/site-packages/pysphere/ZSI/client.py", line 425, in ReceiveSOAP
    'Response is "%s", not "text/xml"' % self.local.reply_headers.type)
TypeError: Response is "text", not "text/xml"

fatal: [<Guest hostname>]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "vsphere_guest"}, "module_stderr": "Traceback (most recent call last):\n  File \"/var/folders/br/2l8z6cfs03sfwszn020z2t5c0000gn/T/ansible_2jRR_o/ansible_module_vsphere_guest.py\", line 1879, in <module>\n    main()\n  File \"/var/folders/br/2l8z6cfs03sfwszn020z2t5c0000gn/T/ansible_2jRR_o/ansible_module_vsphere_guest.py\", line 1754, in main\n    viserver.connect(vcenter_hostname, username, password)\n  File \"/Library/Python/2.7/site-packages/pysphere/vi_server.py\", line 101, in connect\n    request)._returnval\n  File \"/Library/Python/2.7/site-packages/pysphere/resources/VimService_services.py\", line 2172, in RetrieveServiceContent\n    response = self.binding.Receive(RetrieveServiceContentResponseMsg.typecode)\n  File \"/Library/Python/2.7/site-packages/pysphere/ZSI/client.py\", line 540, in Receive\n    self.ReceiveSOAP(**kw)\n  File \"/Library/Python/2.7/site-packages/pysphere/ZSI/client.py\", line 425, in ReceiveSOAP\n    'Response is \"%s\", not \"text/xml\"' % self.local.reply_headers.type)\nTypeError: Response is \"text\", not \"text/xml\"\n", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}

NO MORE HOSTS LEFT *************************************************************
 [WARNING]: Could not create retry file 'CS_FreeBSD.retry'.         [Errno 2] No
such file or directory: ''


PLAY RECAP *********************************************************************
<Guest hostname>          : ok=0    changed=0    unreachable=0    failed=1
```

To fix it, I had added /sdk at the end of V-center URL

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Fixes: #24672

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
lib/ansible/modules/cloud/vmware/vsphere_guest.py

##### ANSIBLE VERSION
```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
ready_for_review

File: lib/ansible/modules/cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -665,9 +646,9 @@ def deploy_template(vsphere_client, guest, resource_pool, template_src, esxi, mo
 
         # Datacenter managed object reference
         dclist = [k for k,
-                 v in vsphere_client.get_datacenters().items() if v == datacenter]
+                  v in vsphere_client.get_datacenters().items() if v == datacenter]
```
Comment (dagwieers) [inline lib/ansible/modules/cloud/vmware/vsphere_guest.py]: I would put this on one line. very weird how this one was split :-)

**Comment by dagwieers** [comment]
AFAICT, this also adds PEP8 compliancy. Maybe you could remove the module from the legacy list ?

**Review by dagwieers — APPROVED** [review]

**Comment by Akasurde** [comment]
Currently, there is no integration test available for vsphere_guest, so this change may not guarantee the stability of vsphere_guest. Closing this PR till the time we have integration test in place for this module as well.

**Comment by Akasurde** [comment]
Depends upon https://github.com/vmware/govmomi/issues/945

**Comment by gundalow** [comment]
As agreed in 2017-12-07 Core meeting, we've done bulk pep8 fixes across all modules in `devel`.

This was done in https://github.com/ansible/ansible/pull/33682 and https://github.com/ansible/ansible/pull/33690 which have been merged into `devel`. because of this you will need to rebase your PR. Thanks in advance.

**Pull request closed without merging**